pkg/crypto/internal/walletapi/wallet_api.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package walletapi
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/internal/cryptoutil"
+)
+
+// CreateDIDOpts holds the options for creating DID
+type CreateDIDOpts struct {
+	ServiceType string
+}
+
+// KeyHolder denotes an object that holds keypairs, indexed by the public key (in base 58)
+type KeyHolder interface {
+	// GetKey gets the keypair associated to the given pubkey
+	GetKey(verkey string) (*cryptoutil.KeyPair, error)
```

Comment:
Should it be GetKey(kid string)? verification key is very specific

Reply:
this will be replaced with GetKeyPairCombo in my PR later.. it will return a struct that has 2 key pairs, one for encryption and one for signature... naming is not important (for now)

Reply:
Renamed these, good point